src/lsm/binary_search.zig

```diff
@@ -339,3 +562,128 @@ test "binary search: random" {
         try test_binary_search.random_search(rng.random(), i);
     }
 }
+
+test "binary search: range" {
```

Comment:
Let's write this in the same style as `binary search: explicit`. Instead of manually `try std.desting.expect` for each result, define a helper functions which takes both the input and the output, and asserts inside. 

Duplication of expectations in the tests makes future refactors painful, so I like to introduce a helper testing function (and it also good to write tests in the same style)